exporter/azuremonitorexporter/config.go

```diff
@@ -22,4 +23,5 @@ type Config struct {
 	ShutdownTimeout        time.Duration                   `mapstructure:"shutdown_timeout"`
 	CustomEventsEnabled    bool                            `mapstructure:"custom_events_enabled"`
 	ExceptionEventsEnabled bool                            `mapstructure:"exception_events_enabled"`
+	ClientConfig           confighttp.ClientConfig         `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
```

Comment:
not sure squash is preferred here?

Reply:
I copied what was done in the prometheusremotewrite exporter [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/db9f5647bce2cdf861d0f14fe8baf9829137d07e/exporter/prometheusremotewriteexporter/config.go#L35C67-L35C74).
The goal is to have the same config at the same level, so I figured this is what we need. If we don't squash, it would introduce an extra level in the config. Ideally, the experience would be the same across different exporters in this repo.

Reply:
will it conflict with existing settings like `endpoint`?

Reply:
You're right, it probably conflicts with `endpoint` at the root. What do you suggest to fix this?

Reply:
@hgaol do you have a suggestion for this?